sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CollectionExpressionsSuite.scala

```diff
@@ -1883,4 +1887,32 @@ class CollectionExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper
       Literal(stringToInterval("interval 1 year"))),
       Seq(Date.valueOf("2018-01-01")))
   }
+
+  test("SPARK-33391: element_at ArrayIndexOutOfBoundsException") {
```

Comment:
Maybe, `SPARK-33386` is better than `SPARK-33391` because this test name has `ArrayIndexOutOfBoundsException`.